integration/docker_test.go

```diff
@@ -240,28 +240,53 @@ func TestDocker(t *testing.T) {
 		require.Equal(t, "1000", strings.TrimSpace(string(out)))
 
 		// Validate that memory limit is being applied to the inner container.
-		out, err = integrationtest.ExecInnerContainer(t, pool, integrationtest.ExecConfig{
+		// First check under cgroupv2 path.
+		if out, err = integrationtest.ExecInnerContainer(t, pool, integrationtest.ExecConfig{
 			ContainerID: resource.Container.ID,
-			Cmd:         []string{"cat", "/sys/fs/cgroup/memory/memory.limit_in_bytes"},
-		})
-		require.NoError(t, err)
-		require.Equal(t, expectedMemoryLimit, strings.TrimSpace(string(out)))
+			Cmd:         []string{"cat", "/sys/fs/cgroup/memory.max"},
+		}); err == nil {
+			require.Equal(t, expectedMemoryLimit, strings.TrimSpace(string(out)))
+		} else { // fall back to cgroupv1 path.
+			out, err = integrationtest.ExecInnerContainer(t, pool, integrationtest.ExecConfig{
+				ContainerID: resource.Container.ID,
+				Cmd:         []string{"cat", "/sys/fs/cgroup/memory/memory.limit_in_bytes"},
+			})
+			require.NoError(t, err)
+			require.Equal(t, expectedMemoryLimit, strings.TrimSpace(string(out)))
```

Comment:
This only gets run when cgroup v2 fails or is unavailable, do we ever expect that to be the case with our supported versions? Since this test code only gets run when cgroup v2 fails, will it ever get run in CI or on our dev machines?